webapp/src/components/sidebar_right/sidebar_right.tsx

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import Scrollbars from 'react-custom-scrollbars';
+import {Theme} from 'mattermost-redux/types/preferences';
+import {makeStyleFromTheme, changeOpacity} from 'mattermost-redux/utils/theme_utils';
+
+import {RHSStates} from '../../constants';
+import GitlabItems from './gitlab_items';
+import {Item} from '../../types/gitlab_items';
+
+interface PropTypes {
+    username: string;
+    org: string;
+    gitlabURL: string;
+    reviews: Item[];
+    unreads: Item[],
+    yourPrs: Item[],
+    yourAssignments: Item[],
+    rhsState: string,
+    theme: Theme,
+    actions:any,
+};
+
+export function renderView(props: PropTypes) {
+    return (
+        <div
+            {...props}
+            className='scrollbar--view'
+        />);
+}
+
+export function renderThumbHorizontal(props: PropTypes) {
+    return (
+        <div
+            {...props}
+            className='scrollbar--horizontal'
+        />);
+}
+
+export function renderThumbVertical(props: PropTypes) {
+    return (
+        <div
+            {...props}
+            className='scrollbar--vertical'
+        />);
+}
+
+
+function mapGitlabItemListToPrList(gilist: Item[]) {
+    if (!gilist) {
+        return [];
+    }
+
+    return gilist.map((pr: Item) => {
+        return {sha: pr.sha, project_id: pr.project_id, iid: pr.iid};
+    });
+}
+
+function shouldUpdateDetails(prs: Item[], prevPrs: Item[], targetState: string, currentState: string, prevState: string) {
+    if (currentState !== targetState) {
+        return false
+    }
+
+    if (currentState !== prevState) {
+        return true;
+    }
+
+    if (prs.length !== prevPrs.length) {
+        return true;
+    }
+
+    for (let i = 0; i < prs.length; i++) {
```

Comment:
what is the logic behind this for loop? like under what conditions the truthy condition should be returned? have this been validated that accomplish the expected logic?

Reply:
This loop is to check if the PR's in the previous props are different from the PR's in current props. It will return true if the PR's are different.